lib/manageiq/api/client/client.rb

```diff
@@ -31,6 +30,25 @@ def load_definitions
         @collections   = load_collections(entrypoint["collections"])
       end
 
+      def update_authentication(auth_options = {})
+        return @authentication unless ManageIQ::API::Client::Authentication.options?(auth_options)
+        saved_auth = @authentication
+        @authentication = ManageIQ::API::Client::Authentication.new(auth_options)
+        begin
+          reconnect
+        rescue => err
+          @authentication = saved_auth
+          raise err
+        end
+        @authentication
+      end
+
+      def reconnect
+        new_connection = ManageIQ::API::Client::Connection.new(self, options.slice(:ssl))
+        @connection = new_connection
```

Comment:
I had it like that, but in case of errors with new credentials when reconnecting I could no longer use the client. In the above, the connection is updated (line 48) only in the case of a successful auth on line 47.